File: Java/src/main/java/BuilderPattern/HouseDirector.java
package BuilderPattern;

import Utils.Constants;

public class HouseDirector extends Director {

    public static final String simpleBuilderType = Constants.BuilderType.simple;

    public static String luxuryBuilderType = Constants.BuilderType.luxury;

    public static String ultraLuxuryBuilderType = Constants.BuilderType.ultraLuxury;

    @Override
    void make(String builderType) {
        switch (builderType){
            case simpleBuilderType:
                this.builder.BuildCeiling();
                this.builder.BuildFloors();
                this.builder.BuildGarage();
                break;
            default:
                throw new IllegalStateException("Unexpected value: " + builderType);
        }
    }
}

File: TS/BuilderPattern/HouseDirector.ts
import Director, { BuildingType } from "./IDirector";

export default class HouseDirector extends Director {
  make(buildingType: BuildingType): void {
    switch (buildingType) {
      case BuildingType.simple:
        this.builder.buildCeiling();
        this.builder.buildFlooring();
        this.builder.buildRooms();
        break;
      case BuildingType.luxury:
        this.builder.buildCeiling();
        this.builder.buildFlooring();
        this.builder.buildPool();
        this.builder.buildRooms();
        break;
      case BuildingType.ultraLuxury:
        this.builder.buildCeiling();
        this.builder.buildFlooring();
        this.builder.buildPool();
        this.builder.buildPorch();
        this.builder.buildRooms();
        break;
      default:
        console.log("Improper Building Type Selected");
        break;
    }
  }
}
